libs/shared/shared_common.c

```diff
@@ -1,8 +1,151 @@
 // Copyright (c) Microsoft Corporation
 // SPDX-License-Identifier: MIT
 
+#include "ebpf_program_types.h"
 #include "ebpf_shared_framework.h"
 
+enum _extension_object_type
+{
+    EBPF_PROGRAM_TYPE_DESCRIPTOR = 0,
+    EBPF_HELPER_FUNCTION_PROTOTYPE,
+    EBPF_PROGRAM_INFO,
+    EBPF_PROGRAM_DATA,
+    EBPF_PROGRAM_SECTION,
+};
+
+// Supported version and sizes of the various extension data structures.
+
+uint16_t _supported_ebpf_extension_version[] = {
+    EBPF_PROGRAM_TYPE_DESCRIPTOR_CURRENT_VERSION,
+    EBPF_HELPER_FUNCTION_PROTOTYPE_CURRENT_VERSION,
+    EBPF_PROGRAM_INFORMATION_CURRENT_VERSION,
+    EBPF_PROGRAM_DATA_CURRENT_VERSION,
+    EBPF_PROGRAM_SECTION_INFORMATION_CURRENT_VERSION,
+};
+
+#define EBPF_PROGRAM_TYPE_DESCRIPTOR_SIZE_0 EBPF_OFFSET_OF(ebpf_program_type_descriptor_t, is_privileged) + sizeof(char)
```

Comment:
nit: Should we define and use a common macro (something like below) for these?
```
#define EBPF_FIELD_OFFSET(s, m) (EBPF_OFFSET_OF(s, m) + sizeof(((s*)0)->m))
```

Reply:
a common macro is a good idea